x-pack/plugins/security_solution/server/endpoint/services/actions.ts

```diff
@@ -135,3 +141,65 @@ const getActivityLog = async ({
 
   return sortedData;
 };
+
+export const getPendingActionCounts = async (
+  esClient: ElasticsearchClient,
+  agentIDs: string[]
+): Promise<EndpointPendingActions[]> => {
+  // retrieve the unexpired actions for the given hosts
+
+  const recentActions = await searchUntilEmpty<EndpointAction>(esClient, {
+    index: AGENT_ACTIONS_INDEX,
+    body: {
+      query: {
+        bool: {
+          filter: [
+            { term: { type: 'INPUT_ACTION' } }, // actions that are directed at agent children
+            { term: { input_type: 'endpoint' } }, // filter for agent->endpoint actions
+            { range: { expiration: { gte: 'now' } } }, // that have not expired yet
+            { terms: { agents: agentIDs } }, // for the requested agent IDs
+          ],
+        },
+      },
+    },
+  });
+
+  // retrieve any responses to those action IDs from these agents
+  const actionIDs = recentActions.map((a) => a.action_id);
+  const responses = await searchUntilEmpty<EndpointActionResponse>(esClient, {
+    index: '.fleet-actions-results',
```

Comment:
Is there a const for this as well that you can pull from fleet?

Reply:
Yeah, there is. Nice catch. will use it